src/bootstrap/builder.rs

```diff
@@ -600,9 +600,20 @@ impl<'a> Builder<'a> {
         //
         // FIXME: the guard against msvc shouldn't need to be here
         if !target.contains("msvc") {
-            let cc = self.cc(target);
-            cargo.env(format!("CC_{}", target), cc)
-                 .env("CC", cc);
+            let ccache = self.config.ccache.as_ref();
+            let ccacheify = |s: &Path| {
+                let ccache = match ccache {
+                    Some(ref s) => s,
+                    None => return s.display().to_string(),
+                };
+                match &ccache[..] {
+                    "ccache" | "sccache" => format!("{} {}", ccache, s.display()),
+                    _ => s.display().to_string(),
```

Comment:
Why we need to special-case `ccache | sccache` here?

Reply:
Unfortunately `cc-rs` isn't [the most general](https://github.com/alexcrichton/cc-rs/blob/a00c415929156aeba5485d7dee27f8fe1d1a69bf/src/lib.rs#L1610) either :(